core-lib/src/main/java/org/idpass/smartscanner/lib/nfc/NFCResult.kt

```diff
@@ -42,4 +48,50 @@ data class NFCResult(
         var taxOrExitRequirements: String? = null,
         var mrzOptional: String? = null,
         var mrzOptional2: String? = null
-)
\ No newline at end of file
+) {
+    companion object {
+
+        fun formatResult(passport: Passport?, mrzInfo: MRZInfo? = null): NFCResult {
+            val personDetails = passport?.personDetails
+            val additionalPersonDetails = passport?.additionalPersonDetails
+            val additionalDocumentDetails = passport?.additionalDocumentDetails
+            // Note: Get proper names
+            // NFCResult.nameOfHolder --> LAST_NAME<<GIVEN_NAMES (RTL for Arabic)
+            // we split nameOfHolder to two parts separated by '<<' (double chevron)
+            val nameParts  = additionalPersonDetails?.nameOfHolder?.split("<<")?.toMutableList()
+            // then first part of nameOfHolder should contain LAST_NAME
+            val surname = nameParts?.get(0)
```

Comment:
@rjmangubat23 The surname can also contain multiple words separated by `<`.

Reply:
yes, the parts may has size more or less than 3 .. so we have to make it dynamically not static